src/setuptools_scm/_entrypoints.py

```diff
@@ -43,27 +59,11 @@ def version_from_entrypoint(
     return None
 
 
-if sys.version_info[:2] < (3, 10):
-    from importlib_metadata import entry_points
-    from importlib_metadata import EntryPoint
-else:
-    from importlib.metadata import entry_points
-    from importlib.metadata import EntryPoint
-
-
-def iter_entry_points(
-    group: str, name: str | None = None
-) -> Iterator[EntrypointProtocol]:
-    eps = entry_points(group=group)
-    res = (
-        eps
-        if name is None
-        else eps.select(  # type: ignore [no-untyped-call]
-            name=name,
-        )
-    )
+def iter_entry_points(group: str, name: str | None = None) -> Iterator[EntryPoint]:
+    eps: EntryPoints = entry_points(group=group)
+    res = eps if name is None else eps.select(name=name)
```

Comment:
I'm not sure that typing works in py38

```
>>> def iter_entry_points(group: str, name: str | None = None) -> Iterator[EntryPoint]:
...    return []
... 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported operand type(s) for |: 'type' and 'NoneType'
```

maybe 

```suggestion
def iter_entry_points(group: str, name: Optional[str] = None) -> Iterator[EntryPoint]:
    eps: EntryPoints = entry_points(group=group)
    res = eps if name is None else eps.select(name=name)
```

Reply:
🤷🏻‍♂️ but TBH -- I'm not sure how this worked before your change

Reply:
all is well, I see the 

```python
from __future__ import annotations
```

in the header.  Carry on!